### PR TITLE
Add AddRequestId property to ClientOptions

### LINKED ISSUE #423: addRequestIds client option
Conforming to [RSC7c](https://docs.ably.io/client-lib-development-guide/features/#RSC7c) and [TO3p](https://docs.ably.io/client-lib-development-guide/features/#TO3p).

┆Issue is synchronized with this [Jira Story](https://ably.atlassian.net/browse/SDK-109) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by sacOO7**
- Added ```AddRequestIds``` option to the ClientOptions
- Fixes #423

### CODE REVIEW

File: src/IO.Ably.Shared/AblyRest.cs [hunk 1]
```diff
@@ -169,11 +178,20 @@ internal async Task<AblyResponse> ExecuteRequest(AblyRequest request)
...
                 }
 
                 throw new AblyException(ex);
             }
+
+            string WrapWithRequestId(string message)
+            {
+                if (requestId != null)
+                {
+                    message = $"RequestId {requestId} : {message}";
```
Comment (marto83) [inline src/IO.Ably.Shared/AblyRest.cs:191]: just do return here. It's good practise not to reassign function parameters

File: src/IO.Ably.Shared/AblyRest.cs [hunk 2]
```diff
@@ -5,6 +5,7 @@
 using System.Net;
 using System.Net.Http;
 using System.Threading.Tasks;
+using Foundation;
```
Comment (marto83) [inline src/IO.Ably.Shared/AblyRest.cs:8]: What is this namespace? 🤔

**Review by marto83 — COMMENTED** [review]
Looks good. Just a couple of small questions.

File: src/IO.Ably.Shared/AblyRest.cs (hunk 2 above)
Comment (sacOO7) [reply src/IO.Ably.Shared/AblyRest.cs:8]: I think It's being used for ```Guid.NewGuid().ToByteArray()```.
Lol, I just remembered, it's a base namespace used in swift.

File: src/IO.Ably.Shared/ClientOptions.cs [hunk 3]
```diff
@@ -328,6 +328,12 @@ public bool UseBinaryProtocol
         /// </summary>
         public bool AutomaticNetworkStateMonitoring { get; set; } = true;
 
+        /// <summary>
+        /// If enabled, every REST request to Ably includes a `request_id` query string parameter.
+        /// This request ID remain the same if a request is retried to a fallback host.
```
Comment (tomkirbygreen) [inline src/IO.Ably.Shared/ClientOptions.cs:333]: Minor: Grammar, `remain` => `remains`.

**Review by marto83 — APPROVED** [review]
Looks good

File: src/IO.Ably.Shared/ClientOptions.cs (hunk 3 above)
Comment (marto83) [reply src/IO.Ably.Shared/ClientOptions.cs]: I've updated it

**Comment by tomkirbygreen** [comment]
Replaced by PR https://github.com/ably/ably-dotnet/pull/530 .

**Pull request closed without merging**